csrc/ir/nodes.cpp

```diff
@@ -425,18 +425,34 @@ std::vector<PolymorphicValue> UnaryOp::evaluate(
     }
 
     Val* bias = nullptr;
+    Val* alpha = nullptr;
+    Val* beta = nullptr;
     // Case 2: Matmul + Bias
     if (MmaOpUtils::matchMatmulBiasCast(
-            this, std::ref(mma_lhs), std::ref(mma_rhs), std::ref(bias))) {
+            this,
+            std::ref(mma_lhs),
+            std::ref(mma_rhs),
+            std::ref(bias),
+            std::ref(alpha),
+            std::ref(beta))) {
       // Inputs to MmaOp are of the shape [M, K, 1] x [1, K, N]
       const auto a =
           ee.evaluate(mma_lhs, known_values).as<at::Tensor>().squeeze(-1);
       const auto b =
           ee.evaluate(mma_rhs, known_values).as<at::Tensor>().squeeze(0);
-      // Bias is of shape [M,].
-      const auto bias_tensor =
-          ee.evaluate(bias, known_values).as<at::Tensor>().unsqueeze(-1);
-      return {at::addmm(bias_tensor, a, b)};
+      auto bias_tensor = ee.evaluate(bias, known_values).as<at::Tensor>();
+
+      if (bias_tensor.dim() != a.dim()) {
```

Comment:
Technically, we could have a bias that is 3D and matmul output that is 2D and add them, relying on implicit broadcasting. So we should probably just explicitly check the case that `bias_tensor.dim() == a.dim()` or `bias_tensor.dim() == a.dim() - 1` else fail the pattern.

Reply:
Only the second condition should occur, right? Matmul operands will be 3D and bias will be 2D.

Reply:
OK yeah I forgot about the broadcast. but we could then have a 4D or 5D "bias" being added to the matmul output. This current pattern would match it and unsqueeze to 5 or 6 dimensions, whereas it should just fail to match in those cases.